### PR TITLE
Update GuidelinesForNewRules.md

### PR DESCRIPTION
**Pull request by mavasani**
Update documentation requirements and steps for creating documentation PRs for new CA rules

### CODE REVIEW

**Comment by mavasani** [comment]
@dotpaul @vatsalyaagrawal @jeffhandley These are the new documentation requirements that we discussed offline. Please review if it looks fine to you. Thanks!

**Comment by mavasani** [comment]
Also tagging @Mikejo5000

File: GuidelinesForNewRules.md [hunk 1]
```diff
@@ -21,4 +21,26 @@
...
+5. Documentation requirements:
+   1. **New CA rule must be documented by the rule author**: Each rule ID `CAxxxx` is automatically assigned the help link https://docs.microsoft.com/visualstudio/code-quality/caxxxx. The documentation for this page is populated from `caxxxx.md` file at https://github.com/MicrosoftDocs/visualstudio-docs/tree/master/docs/code-quality. For example, `CA1000` is documented at [ca1000.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/code-quality/ca1000.md) file. Documenting a new rule is primarily ensuring a PR is send to `MicrosoftDocs` repo to add `caxxxx.md` file for the new rule. Detailed steps are given below.
```
Comment (dotpaul) [inline GuidelinesForNewRules.md:25]: >by the rule author [](start = 39, length = 18)

Does it have to be "by the rule author"? I think it'd be fine to omit that part.

Comment (mavasani) [inline GuidelinesForNewRules.md:25]:
```suggestion
   1. **New CA rule must be documented**: Each rule ID `CAxxxx` is automatically assigned the help link https://docs.microsoft.com/visualstudio/code-quality/caxxxx. The documentation for this page is populated from `caxxxx.md` file at https://github.com/MicrosoftDocs/visualstudio-docs/tree/master/docs/code-quality. For example, `CA1000` is documented at [ca1000.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/code-quality/ca1000.md) file. Documenting a new rule is primarily ensuring a PR is send to `MicrosoftDocs` repo to add `caxxxx.md` file for the new rule. Detailed steps are given below.
```

File: GuidelinesForNewRules.md [hunk 2]
```diff
@@ -21,4 +21,26 @@
...
+5. Documentation requirements:
+   1. **New CA rule must be documented**: Each rule ID `CAxxxx` is automatically assigned the help link https://docs.microsoft.com/visualstudio/code-quality/caxxxx. The documentation for this page is populated from `caxxxx.md` file at https://github.com/MicrosoftDocs/visualstudio-docs/tree/master/docs/code-quality. For example, `CA1000` is documented at [ca1000.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/code-quality/ca1000.md) file. Documenting a new rule is primarily ensuring a PR is send to `MicrosoftDocs` repo to add `caxxxx.md` file for the new rule. Detailed steps are given below.
```
Comment (Mikejo5000) [inline GuidelinesForNewRules.md:25]:
```suggestion
   1. **New CA rule must be documented**: Each rule ID `CAxxxx` is automatically assigned the help link https://docs.microsoft.com/visualstudio/code-quality/caxxxx. The documentation for this page is populated from `caxxxx.md` file at https://github.com/MicrosoftDocs/visualstudio-docs/tree/master/docs/code-quality. For example, `CA1000` is documented at [ca1000.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/code-quality/ca1000.md) file. Documenting a new rule is primarily ensuring a PR is sent to `MicrosoftDocs` repo to add `caxxxx.md` file for the new rule. Detailed steps are given below.
```

**Comment by Mikejo5000** [comment]
Looks good!

**Comment by mavasani** [comment]
Thanks!

**Pull request merged**